File: OPEN_DMS/Models/Document.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace OPEN_DMS.Models
{
    public partial class Document
    {
        public int Id { get; set; }
        public string FileName { get; set; }
        public string Extension { get; set; }
        public float Size { get; set; }
        public int TeamId { get; set; }
        public DateTime InsertionDate { get; set; }
        public string PathAlternative { get; set; }
        public string CommentDetail { get; set; }
        public string DistinctDetail { get; set; }
        public int IdUser { get; set; }
        public bool Disabled { get; set; }
    }
}

File: OPEN_DMS/ClientApp/src/app/classes/document.ts
import { Deserializeble } from "../interfaces/deserializable";

export class DocumentFile implements Deserializeble{

    constructor(
        public Id: number,
        public FileName: string,
        public Extension: string,
        public Size: string,
        public TeamId: number,
        public EntityId: number,
        public InsertionDate: Date,
        public PathAlternative: string,
        public CommentDetail: string,
        public DistinctDetail: string,
        public IdUser: number,
        public Disabled: boolean
    ){}
    
    deserialize(input: any): this {
        Object.assign(this, input);
        return this;
    }
}


export class DocumentFileIcon implements DocumentFile{
    constructor(
        public Id: number,
        public FileName: string,
        public Extension: string,
        public Size: string,
        public TeamId: number,
        public EntityId: number,
        public InsertionDate: Date,
        public PathAlternative: string,
        public CommentDetail: string,
        public DistinctDetail: string,
        public IdUser: number,
        public icon: string,
        public Disabled: boolean
    ){}
    deserialize(input: any): this {
        Object.assign(this, input);
        return this;
    }
}